res/blend.fs.glsl

```diff
@@ -8,8 +6,10 @@ uniform sampler2D sDiffuse;
 uniform sampler2D sMask;
 uniform vec4 uBlendParams;
 
-varying vec2 vColorTexCoord;
-varying vec2 vMaskTexCoord;
+IN_VARYING vec2 vColorTexCoord;
```

Comment:
I think what we should do is introduce 2 new resource files, something like es2_common.glsl and gl3_common.glsl. These could define a function for sampling a texture, and also define all the attributes, varyings etc. Then the runtime code can prefix the shader compiles with the appropriate file. This means we won't need any preprocessor tricks, which I've found flaky in several real world GL drivers. It will also simplify the shaders since we'll have one place to declare all varyings etc (it doesn't matter if you declare unused varyings, attributes etc - they'll get stripped out).